packages/transactions/src/decompile-transaction.ts

```diff
@@ -58,6 +59,54 @@ function getAccountMetas(message: CompiledMessage): IAccountMeta[] {
     return accountMetas;
 }
 
+export type LookupTables = { [lookupTableAddress: Address]: Address[] };
+
+function getAddressLookupMetas(
+    transactionLookups: ReturnType<typeof getCompiledAddressTableLookups>,
+    knownLookups: LookupTables,
+): IAccountLookupMeta[] {
+    // check that all message lookups are known
+    const transactionLookupAddresses = transactionLookups.map(l => l.lookupTableAddress);
+    const missing = transactionLookupAddresses.filter(a => knownLookups[a] === undefined);
+    if (missing.length > 0) {
+        const missingAddresses = missing.join(', ');
+        // TODO: coded error.
+        throw new Error(`Transaction includes missing address lookup tables: [${missingAddresses}]`);
+    }
+
+    const readOnlyMetas: IAccountLookupMeta[] = [];
+    const writableMetas: IAccountLookupMeta[] = [];
+
+    // we know that for each lookup, knownLookups[lookup.lookupTableAddress] is defined
+    for (const lookup of transactionLookups) {
+        const addresses = knownLookups[lookup.lookupTableAddress];
+
+        const highestIndex = Math.max(...lookup.readableIndices, ...lookup.writableIndices);
+        if (highestIndex >= addresses.length) {
+            // TODO coded error
+            throw new Error(`Cannot look up index ${highestIndex} in lookup table ${lookup.lookupTableAddress}`);
```

Comment:
Nit: keep going with this? Explain that there being an index greater than the one in the `addressesByLookupTableAddress` might imply that they supplied an _old_ version of the lookup table – that it might since have been extended with _more_ addresses?

Reply:
Thanks! I added this error mostly as a just-in-case and wasn't sure what would cause it, so this is really helpful. I've added ` The lookup table may have been extended since the addresses provided were retrieved.` to the end.